crates/lib/src/api.rs

```diff
@@ -311,3 +312,42 @@ pub async fn retrieve_results(
         .map(ExecutionResults::from)
         .ok_or_else(|| GrpcClientError::ResponseEmpty("Controller Job Execution Results".into()))
 }
+
+/// Use the qcs api client to fetch all available quantum processors.
+/// Will fail if more than 10 pages are requested without all names
+/// being found.
+pub async fn list_quantum_processor_names(
+    client: &Qcs,
+) -> Result<Vec<String>, Box<dyn std::error::Error + Send + Sync + 'static>> {
+    let mut quantum_processors = vec![];
+    let mut page_token = None;
+
+    // sanity check
+    let mut calls = 0;
+    loop {
```

Comment:
Why not a `for` loop? `for calls in (0..10)` Can error on `page_token.is_some()` instead.

But also, do we expect to ever have 10+ pages of results?

Reply:
No (or even more than one), but I figured it'd be some protection against endlessly spinning against the API incase there's a bug that makes the pages loop or something. I may just remove pagination.

Reply:
I think it would be more conventional to just accept a timeout value, which also accounts for more failure modes. Accounting for one particular API failure mode in the client, in a way that introduces a new failure mode, seems unwise to me. Imagine that at one point we do have >1000 QPU IDs (maybe because our tests create them, and an internal user is running this) - now this version of the client will permanently fail until we delete some of those. With a user-configurable timeout, there's at least a client-side workaround to keep this client version usable.

Reply:
Removed the pagination.